go.mod

```diff
@@ -2,11 +2,14 @@ module github.com/kairos-io/kairos-challenger
 
 go 1.20
 
+// The address is github.com/kairos-io/kairos-agent but the go module is called differently
+replace github.com/kairos-io/kairos/v2 v2.1.3 => github.com/kairos-io/kairos-agent/v2 v2.1.3
```

Comment:
Im not sure how this can be fixed in the go.mod?

Best solution would be to move the collector to the sdk, but Its too tied into the install config? I would have expected it to be more losely coupled, like you pass an interface and the dirs and the collector unmarshalls into that interfacte.

Maybe we need a type for our configs that can be passed to the collector?

Reply:
The collector pkg should not be aware of any specific config. They idea was that it can parse a number of sources (cmdline, yamls in directories, remote yamls etc) and it will construct one yaml out of them by merging them all together.

It should be generic enough to be moved to a completely separate library. Maybe we introduced some dependency at some point that I'm not aware of, but the above was the original plan.

Reply:
IIRC from yesterday, collector imports config directly, which is the Install config so it's directly tied to it.

Hence my suggestion to move it to an interface or something more generic that can be plugged on

Reply:
I don't see any import: https://github.com/kairos-io/kairos-agent/blob/d3f9f169da98e069d0cbddb5fe21669121130f86/pkg/config/collector/collector.go#L6-L21

Reply:
Ah yeah, it's on the tests and I didn't want to bother reworking them 😂

Reply:
asaict, the tests import the kairos config, just for this function: https://github.com/kairos-io/kairos-agent/blob/d3f9f169da98e069d0cbddb5fe21669121130f86/pkg/config/collector/collector_test.go#L371 . `Scan` just needs a function with the signature of `FilterKeys`, we can create one in the tests to avoid importing anything.